crates/ethereum/node/tests/e2e/rpc.rs

```diff
@@ -0,0 +1,109 @@
+use crate::utils::eth_payload_attributes;
+use alloy_eips::calc_next_block_base_fee;
+use alloy_primitives::U256;
+use alloy_provider::{network::EthereumWallet, Provider, ProviderBuilder};
+use rand::{rngs::StdRng, Rng, SeedableRng};
+use reth_chainspec::{ChainSpecBuilder, MAINNET};
+use reth_e2e_test_utils::setup_engine;
+use reth_node_ethereum::EthereumNode;
+use std::sync::Arc;
+
+alloy_sol_types::sol! {
+    #[sol(rpc, bytecode = "6080604052348015600f57600080fd5b5060405160db38038060db833981016040819052602a91607a565b60005b818110156074576040805143602082015290810182905260009060600160408051601f19818403018152919052805160209091012080555080606d816092565b915050602d565b505060b8565b600060208284031215608b57600080fd5b5051919050565b60006001820160b157634e487b7160e01b600052601160045260246000fd5b5060010190565b60168060c56000396000f3fe6080604052600080fdfea164736f6c6343000810000a")]
+    contract GasWaster {
+        constructor(uint256 iterations) {
+            for (uint256 i = 0; i < iterations; i++) {
+                bytes32 slot = keccak256(abi.encode(block.number, i));
+                assembly {
+                    sstore(slot, slot)
+                }
+            }
+        }
+    }
+}
+
+#[tokio::test]
+async fn test_fee_history() -> eyre::Result<()> {
+    reth_tracing::init_test_tracing();
+
+    let seed: [u8; 32] = rand::thread_rng().gen();
+    let mut rng = StdRng::from_seed(seed);
+    println!("Seed: {:?}", seed);
+
+    let chain_spec = Arc::new(
+        ChainSpecBuilder::default()
+            .chain(MAINNET.chain)
+            .genesis(serde_json::from_str(include_str!("../assets/genesis.json")).unwrap())
+            .cancun_activated()
+            .build(),
+    );
+
+    let (mut nodes, _tasks, wallet) =
+        setup_engine::<EthereumNode>(1, chain_spec.clone(), false, eth_payload_attributes).await?;
```

Comment:
also, should we wrap this into a helper type

https://github.com/paradigmxyz/reth/blob/5241586fb58a726d7e3c9f5570c50e4416ba9ac9/crates/e2e-test-utils/src/lib.rs#L113-L113

I assume eventually we could use some utils that need to operate on all nodes for example